Source/WebCore/page/csp/ContentSecurityPolicy.h

```diff
@@ -290,6 +294,7 @@ class ContentSecurityPolicy final : public CanMakeThreadSafeCheckedPtr<ContentSe
     mutable std::optional<ContentSecurityPolicyResponseHeaders> m_cachedResponseHeaders;
     bool m_isHeaderDelivered { false };
     ContentSecurityPolicyModeForExtension m_contentSecurityPolicyModeForExtension { ContentSecurityPolicyModeForExtension::None };
+    std::unique_ptr<HashAlgorithmSetCollection> m_hashesToReport;
```

Comment:
Since this never needs to be resized after being created it should be a `FixedVector` (not sure I remembered that name right) and then doesn't need to be stored in a `unique_ptr` since an empty one is just a null pointer I think. The main feature of `Vector` is its ability to be efficiently resized.

Reply:
Apologies, but I don't think I understand. In ContentSecurityPolicy::hashesToReport, we are appending pairs to the initially empty vector (so IIUC, it does get resized from its initial empty state). Are you suggesting a different approach?

Reply:
When creating we can use something resizable like a `Vector` with inline capacity then copy into a fixed vector to keep it around. `Vector` pays extra memory so it can be efficiently resized later, a capability we don't need. And it's particularly wasteful to use a Vector stored on the heap.

Reply:
Got it! done!